ledger/ledgercore/statedelta.go

```diff
@@ -263,26 +263,34 @@ func (ad NewAccountDeltas) ModifiedAccounts() []basics.Address {
 		result[i] = ad.accts[i].Addr
 	}
 
-	// consistency check: ensure all addresses in params/holdings/states are also in base account
-	// TODO: remove after the schema switch
-	for aapp := range ad.appParamsCache {
-		if _, ok := ad.acctsCache[aapp.Address]; !ok {
-			panic(fmt.Sprintf("account app param delta: addr %s not in base account", aapp.Address))
+	// consistency check: ensure all addresses for deleted params/holdings/states are also in base accounts
```

Comment:
we might want to delete these loops to get some perf but it would require splitting `Put[Asset|App][Params|State|Holding]` into Create and Update methods to have such check in Create methods.